gammapy/modeling/models/core.py

```diff
@@ -372,6 +373,35 @@ def to_dict(self, full_output=False):
         else:
             return {"components": models_data}
 
+    def to_parameters_table(self):
+        """Convert Models parameters to an astropy Table."""
+        pars = self.parameters.to_table()
+        rows = []
+        #Warning: splitting of parameters will break is source name has a "." in its name.
+        for name in self.parameters_unique_names:
+            row = {}
+            parts = name.split(".")
+            row["model"], row["type"], _ = parts
+            rows.append(row)
+        table = table_from_row_data(rows)
+        table = hstack([table, pars])
+        self._table_cached = table
+        return  table
+
+    def set_parameters_from_table(self, t):
+        """Update Models from an astropy Table."""
+#        self._check_non_editable_columns(t, self._table_cached)
+        parameters_dict = [dict(zip( t.colnames, row)) for row in t]
+        #Astropy Table has masked values for NaN. Replacing with np.nan in dict.
+        for k, data in enumerate(parameters_dict):
```

Comment:
Can you check, whether this part could be moved to the attribute setters of the `Parameter` object? E.g cast to `bool` in https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/parameter.py#L204...

Reply:
I managed to move the test for `frozen` parameter being a bool and the replacement of np masked values in the setter.